### PR TITLE
Implement basic YT Music connector.

### PR DESCRIPTION
**Pull request by jaccarmac**
See #1641 

I've deliberately waited to add tests because this is incomplete. The way YT has set up this new page is a bit strange. On the upside, the elements don't move around as you navigate so we can lock onto the bar. However, now-playing status and artist/album splitting are based on hardcoding strings at the moment. Now-playing especially may not work in non-English locales. Also the way artists are handled is a little weird, features get put in the artist field on the page but are also included in track titles. Since artists are hyperlinked it is possible to pull apart the main artist from their features, but I have not done so at this point.

TL;DR: No tests yet, C+C accepted and encouraged!

### CODE REVIEW

**Comment by jaccarmac** [comment]
The unrecognized track may be buggy. I noticed a not recognized popup appear but on switching to the window the track info had been parsed successfully. I'll get back to it when I have time but anyone else who wants to try it make sure to enable that setting to smoke test.

**Comment by jaccarmac** [comment]
That check failure smelled like an eslint version conflict or something. Real weird, everything linted on my side but apparently Grunt changed the formatting.

**Comment by inverse** [comment]
A good start? how are you finding using it in practice? I think we need to also cover cases. e.g. 

- When it's just playing youtube videos, it didn't recognise things for me
- Skipping ads

**Comment by inverse** [comment]
Anymore work on this?

**Comment by jaccarmac** [comment]
Not yet. I need to come back to it but haven't had a chance as it's mostly working for me so far.

**Comment by jaccarmac** [comment]
Follow the contribution guide in this repo but check out my branch instead of master.

**Comment by inverse** [comment]
@jaccarmac if you could:

- Resolve the merge conflicts
- Figure out the normal music detection
- Ad playback prevention scrobbling 

then we should be good to go!

**Comment by jaccarmac** [comment]
Won't have time this weekend, will try to carve some out next week.

**Comment by jaccarmac** [comment]
All those things should be addressed. Let me know what things look like on your end, I can't easily test the advertisement experience.

**Comment by jaccarmac** [comment]
@MajorWorkAlot This version should deal with your link more sanely. Still not sure exactly how to determine when to apply the YT filter set to a title.

@inverse Your Boards of Canada video scrobbles properly for me now.
